distributed-tracing-smaug

sallyom opened the pull request:
@quaid 
@HumairAK 

as promised!

sallyom committed:
distributed-tracing-smaug

Signed-off-by: Sally O'Malley <somalley@redhat.com>

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **humairak** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @humairak` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fblog).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/blog/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["humairak"]} -->

sesheta commented:
@sallyom: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
aicoe-ci/prow/pre-commit | f59cfc6095864aea34d0f876bd0fc5f17bad510c | [link](https://prow.operate-first.cloud/view/s3/ci-prow/prow-logs/pr-logs/pull/operate-first_blog/21/pre-commit/1546544420113879040) | true | `/test pre-commit`

[Full PR test history](https://prow.operate-first.cloud/pr-history?org=operate-first&repo=blog&pr=21). [Your PR dashboard](https://prow.operate-first.cloud/pr?query=is%3Apr%20state%3Aopen%20author%3Asallyom). Please help us and [open an issue](https://github.com/operate-first/blog/issues?q=is:issue+is:open) when you hit one in your PR.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

HumairAK commented on content/posts/rh-distributed-tracing/index.mdx:
```diff
@@ -0,0 +1,236 @@
+### Red Hat Distributed Tracing Data Collection
+
+The Red Hat Distributed Tracing Data Collection operator is an opinionated OpenTelemetry Collector Operator. It is easy to install from
+the OperatorHub as an OpenShift administrator. Once running, regular users can create its custom resources in their own projects. The operator
```
> It is easy to install from the OperatorHub as an OpenShift administrator

Maybe mention that it is easy to install _via olm_? and maybe link to:  
https://docs.openshift.com/container-platform/4.10/operators/admin/olm-adding-operators-to-cluster.html#olm-installing-from-operatorhub-using-web-console_olm-adding-operators-to-a-cluster

HumairAK commented on content/posts/rh-distributed-tracing/index.mdx:
```diff
@@ -0,0 +1,236 @@
+The Red Hat Distributed Tracing Data Collection operator is an opinionated OpenTelemetry Collector Operator. It is easy to install from
+the OperatorHub as an OpenShift administrator. Once running, regular users can create its custom resources in their own projects. The operator
+watches all namespaces in the `Smaug` cluster for namespaced custom resources `OpenTelemetryCollector` and `Instrumentation`.
+In a cluster with this operator running, these resources can be found and created by navigating to the `API Explorer` in the cluster console.
```
You can also access these from the left navigation menu `Installed Operators > <operat-name>` 
I think this is the primarily method end-users are expected to follow, and api explorer is generally the alternative

sallyom replied:
updated! will need a new tag if it looks good, and thank you!

HumairAK commented on content/posts/rh-distributed-tracing/index.mdx:
```diff
@@ -0,0 +1,236 @@
+in the Smaug cluster `otel-dev` project. The application includes a simple web server, a service that exposes port 8080 of the web
+server pod, and a route that exposes the service. View the single yaml file
+[here](https://raw.githubusercontent.com/sallyom/golang-ex/master/opentelemetry-sample-app/sample-app.yaml).
+To deploy this application in your own cluster, first create an `otel-dev` namespace or project (or, substitute `otel-dev` everywhere for your namespace).
+and then apply the sample-app yaml file. Since Smaug is an OpenShift cluster, here we use the command-line tool `oc`. However,
```
```suggestion
To deploy this application in your own cluster, first create an `otel-dev` namespace or project (or, substitute `otel-dev` everywhere for your namespace)
and then apply the sample-app yaml file. Since Smaug is an OpenShift cluster, here we use the command-line tool `oc`. However,
```

HumairAK commented:
/lgtm

This is a great read! 

@quaid can you approve this once you've had a look as well?

sesheta commented:
New changes are detected. LGTM label has been removed.

sesheta commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **humairak** after the PR has been reviewed.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fblog).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/operate-first/blog/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["humairak"]} -->

sallyom commented:
@quaid I'll make this small fix here: https://github.com/operate-first/blog/pull/21#discussion_r920141900

quaid commented on content/posts/rh-distributed-tracing/index.mdx:
```diff
@@ -0,0 +1,240 @@
+OpenTelemetry (OTLP), end-users and developers are empowered to choose a telemetry backend that suits their needs. It is vendor neutral,
+open source, and is supported by industry leaders in the observability space. Perhaps you've been planning to add OTLP
+instrumentation to your applications. It is relatively straight-forward and usually involves a few lines of code. The more complicated issue is how
+to collect, export, and visualize the generated data. This post describes recent additions to the Operate-First cloud that have made it easy
```
```suggestion
to collect, export, and visualize the generated data. This post describes recent additions to the Operate First community cloud that have made it easy
```

quaid commented on content/posts/rh-distributed-tracing/index.mdx:
```diff
@@ -0,0 +1,240 @@
+to collect, export, and visualize the generated data. This post describes recent additions to the Operate-First cloud that have made it easy
+for users to collect and analyze OTLP.
+
+Operate-First hosts a [community cloud](https://www.operate-first.cloud/community-cloud/).
+To learn more about the cloud and how you can access it, visit the [onboarding docs](https://www.operate-first.cloud/getting-started).
```
```suggestion
To learn more about the Operate First [community cloud](https://www.operate-first.cloud/community-cloud/) and how you can access it, visit the [onboarding docs](https://www.operate-first.cloud/getting-started).
```

quaid commented on content/posts/rh-distributed-tracing/index.mdx:
```diff
@@ -0,0 +1,240 @@
+To learn more about the cloud and how you can access it, visit the [onboarding docs](https://www.operate-first.cloud/getting-started).
+If, like me, you're a developer wishing to deploy your applications and perform testing in a production environment,
+you might start [here](https://www.operate-first.cloud/developer). From there you can onboard your Kubernetes based cloud-native workloads with help
+from the Operate-First community.
```
```suggestion
from the Operate First community.
```

quaid commented on content/posts/rh-distributed-tracing/index.mdx:
```diff
@@ -0,0 +1,240 @@
+you might start [here](https://www.operate-first.cloud/developer). From there you can onboard your Kubernetes based cloud-native workloads with help
+from the Operate-First community.
+
+Recently, two operators were deployed on the Operate-First `Smaug` cluster that enable collection of OTLP data from
```
```suggestion
Recently, two operators were deployed on the Operate First `Smaug` cluster that enable collection of OTLP data from
```

quaid commented on content/posts/rh-distributed-tracing/index.mdx:
```diff
@@ -0,0 +1,240 @@
+Recently, two operators were deployed on the Operate-First `Smaug` cluster that enable collection of OTLP data from
+any application. The operators are `Red Hat Distributed Tracing Platform` and `Red Hat Distributed Tracing Data Collection`.
+The former includes a Jaeger operator for visualizing OTLP data and the latter includes the OpenTelemetry operator that
+manages namespaced OpenTelemetryCollectors to collect OTLP. These and many other Kubernetes operators are available from the
```
```suggestion
manages namespaced OpenTelemetryCollectors to collect OTLP. These and many other Kubernetes Operators are available from the
```

quaid commented on content/posts/rh-distributed-tracing/index.mdx:
```diff
@@ -0,0 +1,240 @@
+manages namespaced OpenTelemetryCollectors to collect OTLP. These and many other Kubernetes operators are available from the
+[Certified OpenShift Operators Hub](https://catalog.redhat.com/software/operators/explore). This post describes how to connect your
+applications and workloads to these distributed tracing operators to reach your observability goals. You might bring your instrumented applications
+to Operate-First, or you might use this post to collect OTLP data from another K8s cluster.
```
```suggestion
to Operate First, or you might use this post to collect OTLP data from another K8s cluster.
```

sallyom committed:
Update content/posts/rh-distributed-tracing/index.mdx

Co-authored-by: Karsten Wade <quaid@iquaid.org>

schwesig committed:
Every sentence in a single line

Following our approach to have, for an easier review process, every sentence (period) in it's own line.

sallyom committed:
Merge pull request #1 from schwesig/distributed-tracing

Every sentence in a single line

sallyom commented:
@quaid I've merged @schwesig 's commit to format each sentence in a new line - ready for a re-review and thanks!

schwesig commented:
/lgtm

schwesig commented:
/users documentation
/sig community
/kind handbook
/kind website
/kind documentation
/kind experience

quaid commented:
/approve

Thanks for the reminder @sallyom

sesheta commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/operate-first/blog/pull/21#issuecomment-1191808703" title="Approved">quaid</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=operate-first%2Fblog).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/operate-first/blog/blob/main/OWNERS)~~ [quaid]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
